lib/AST/Identifier.cpp

```diff
@@ -28,8 +28,9 @@ raw_ostream &llvm::operator<<(raw_ostream &OS, Identifier I) {
 }
 
 raw_ostream &llvm::operator<<(raw_ostream &OS, DeclBaseName I) {
-  // TODO: Handle special names
-  return OS << I.getIdentifier();
+  if (I.empty())
+    return OS << "_";
```

Comment:
I don't think `<<` should behave differently from `userFacingStr` (or whatever). Either they should both insert the "_" or neither should.